test/quantization/core/test_quantized_tensor.py

```diff
@@ -1049,18 +1055,19 @@ def test_qtensor_masked_fill(self):
 
         # the above loop does the same test as test_qtensor_fill
         # now we will check masked_fill for subset of indices
-        mask = torch.randint(0, 2, (numel, ))
+        mask = torch.randint(0, 2, (numel, )).to(device)
         mask = mask.bool()
-        x = torch.rand(numel)
+        x = torch.rand(numel).to(device)
         qx = torch.quantize_per_tensor(x, scale=scale, zero_point=zero_point, dtype=qtype)
         for qtype, fill_with in itertools.product(types, fills):
             q_masked_fill = qx.clone()
             q_masked_fill.masked_fill_(mask, fill_with)
             ref = qx.clone()
+
             for i in range(numel):
                 if mask[i]:
                     # this assignment doesn't end up calling masked_fill, allowing us to compare the different implementations
-                    ref[i] = fill_with
+                    ref[i] = torch.Tensor([fill_with]).to(device)
```

Comment:
I didn't want to do this, but otherwise I think it creates the tensor on the wrong device. I think when I'm writing index_put I might be able to fix that, since I think that's what this line is calling? 

Please LMK if you have any better ideas for this.

Reply:
lso, I think torch.Tensor is outdated, is it equivalent to `torch.tensor([fill_with], device = device)`?

Reply:
`torch.Tensor` is preferred over `torch.tensor`. 

@jcaip I think what you did is the correct approach to create a scalar on cuda. I recall doing that for index_put

Reply:
my impression is also we should use `torch.tensor`: https://pytorch.org/docs/stable/tensors.html#torch.Tensor

Reply:
@jerryzh168  do you mean lowecase or capital `t`? you linked `Tensor` but mentioned `tensor`.

Reply:
just checked. yeah it is `torch.tensor` that's preferred. sorry about that. 
`torch.tensor(5)` and `torch.Tensor(5)` produce different results, and the former is what's aligned with `np.array(5)`

Reply:
yeah lower case `t` is preferred I think, linking the `Tensor` doc since it talks about how to create a `Tensor` in different scenarios